### ISSUE TITLE
TypeError after clicking "Confirm" in showSimplePickerLocation

### ISSUE THREAD

**Issue by pennyfea** [liodali/osm_flutter#566]
Platform: Flutter web

When using version `^1.3.4`, a TypeError occurs after clicking the "Confirm" button in the showSimplePickerLocation dialog. The error message indicates that the returned object is of type LegacyJavaScriptObject, which is not a subtype of Map<String, double>?. This issue prevents the proper retrieval of the map center.

Error Message:
```
DartError: TypeError: Instance of 'LegacyJavaScriptObject': type 'LegacyJavaScriptObject' is not a subtype of type 'Map<String, double>?'
dart-sdk/lib/_internal/js_dev_runtime/private/ddc_runtime/errors.dart 296:3  throw_
errors.dart:296
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1405:3                         _failedAsCheck
rti.dart:1405
dart-sdk/lib/_internal/js_shared/lib/rti.dart 1395:3                         _generalNullableAsCheckImplementation
rti.dart:1395
packages/flutter_osm_web/src/mixin_web.dart 435:20                           getMapCenter
mixin_web.dart:435
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 45:50           <fn>
async_patch.dart:45
dart-sdk/lib/async/zone.dart 1661:54                                         runUnary
zone.dart:1661
dart-sdk/lib/async/future_impl.dart 163:18                                   handleValue
future_impl.dart:163
dart-sdk/lib/async/future_impl.dart 861:44                                   handleValueCallback
future_impl.dart:861
dart-sdk/lib/async/future_impl.dart 890:13                                   _propagateToListeners
future_impl.dart:890
dart-sdk/lib/async/future_impl.dart 666:5                                    [_completeWithValue]
future_impl.dart:666
dart-sdk/lib/async/future_impl.dart 736:7                                    callback
future_impl.dart:736
dart-sdk/lib/async/schedule_microtask.dart 40:11                             _microtaskLoop
schedule_microtask.dart:40
dart-sdk/lib/async/schedule_microtask.dart 49:5                              _startMicrotaskLoop
schedule_microtask.dart:49
dart-sdk/lib/_internal/js_dev_runtime/patch/async_patch.dart 181:7           <fn>
```

Code Sample:
```
GeoPoint? selectedPoint = await showSimplePickerLocation(
  context: context,
  contentPadding: const EdgeInsets.all(8.0),
  isDismissible: true,
  title: "Select store location",
  textConfirmPicker: "Confirm",
  initPosition: GeoPoint(
      latitude: position.latitude,
      longitude: position.longitude),
);
```

Expected Behavior:
The selected location should be returned as a GeoPoint without errors.

Actual Behavior:
A TypeError occurs due to the returned object being an instance of LegacyJavaScriptObject instead of a Map<String, double>?, causing a failure in retrieving the map center.

Proposed Fix:
The error stems from improper casting of the returned JavaScript object. A type check should be added to safely cast the object as a Map<String, double>.

Suggested fix, not tested:

```
Future<GeoPoint> getMapCenter() async {
  final jsObject = (await interop.centerMap(mapIdMixin.toJS).toDart).dartify();

  if (jsObject is Map) {
    final mapCenterPoint = jsObject.map<String, double>((key, value) {
      if (key is String && value is num) {
        return MapEntry(key, value.toDouble());
      }
      throw Exception("Invalid data type in map");
    });

    return GeoPoint.fromMap(mapCenterPoint);
  }

  throw Exception("web osm: error to get center geopoint");
}
```

**Comment by liodali** [comment]
i will do hotfix today

**Comment by liodali** [comment]
new verions is published `1.3.5`
